lib/game_play.rb

```diff
@@ -13,29 +13,47 @@ class GamePlay
   def initialize(ui)
     @board = Board.new
     @ui = ui
+    @VALUE_MIN = 1
   end
 
   def board_size
-    @ui.board_size
-    selection = get_user_selection_between(3, 4)
+    default_size = 1
+
+    @ui.display_menu_board(default_size)
+    menu_size = @ui.menu_board_size
+    selection = get_user_value(@VALUE_MIN, menu_size, default_size)
+
+    selection += 2
```

Comment:
Why can't `@ui.menu_board_size` return the _actual_ size a user requested? I would treat the 1 and 2 as detail to what it means to have a board size of of 3. In here you're asking the UI to return the requested board size. I think it's fair to assume `@ui` to return the real value and not something that is specific to the currently used user interface.